File: frontend/src/lib/api/spotify.ts
import { apiFetch } from "./client";

export interface SpotifyWidgetResponse {
  track: string;
  artist: string;
  album: string;
  progress_sec: number;
  duration_sec: number;
  is_playing: boolean;
  is_live: boolean;
}

export const spotifyApi = {
  getWidget: () => apiFetch<SpotifyWidgetResponse>("/api/v1/spotify/widget"),
  status: () =>
    apiFetch<{ provider: string; connected: boolean }>("/api/v1/spotify/status"),
  connect: () => apiFetch<{ url: string }>("/api/v1/spotify/connect"),
};

File: backend/app/schemas/spotify.py
from pydantic import BaseModel


class SpotifyWidgetData(BaseModel):
    track: str
    artist: str
    album: str
    progress_sec: int
    duration_sec: int
    is_playing: bool
    # Whether this is real synced data (True) or the demo fallback (False).
    is_live: bool = False


class ConnectionStatus(BaseModel):
    provider: str
    connected: bool
